src/RestrictedPython/Limits.py

```diff
@@ -31,7 +31,7 @@ def limited_range(iFirst, *args):
     if iLen < 0:
         iLen = 0
     if iLen >= RANGELIMIT:
-        raise ValueError('range() too large')
+        raise OverflowError('To be created range() object would be to large, in RestrictedPython we only allow ' + str(RANGELIMIT) + ' elements in a range.')  # NOQA: E501
```

Comment:
`OverflowError` is not a subclass of `ValueError` but of `ArithmeticError` (see https://docs.python.org/3/library/exceptions.html#exception-hierarchy), so this changes the behaviour of the function.

Additionally I think it looks better when breaking the too long line and when using `.format()`.